subprojects/core/src/main/java/org/gradle/configuration/project/BuildScriptProcessor.java

```diff
@@ -18,24 +18,39 @@
 import org.gradle.api.internal.project.ProjectInternal;
 import org.gradle.configuration.ScriptPlugin;
 import org.gradle.configuration.ScriptPluginFactory;
+import org.gradle.internal.event.ListenerManager;
+import org.gradle.internal.resource.local.FileResourceListener;
 import org.gradle.internal.time.Time;
 import org.gradle.internal.time.Timer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.File;
+
 public class BuildScriptProcessor implements ProjectConfigureAction {
     private static final Logger LOGGER = LoggerFactory.getLogger(BuildScriptProcessor.class);
     private final ScriptPluginFactory configurerFactory;
+    private final ListenerManager listenerManager;
 
-    public BuildScriptProcessor(ScriptPluginFactory configurerFactory) {
+    public BuildScriptProcessor(
+        ScriptPluginFactory configurerFactory,
+        ListenerManager listenerManager
+    ) {
         this.configurerFactory = configurerFactory;
+        this.listenerManager = listenerManager;
     }
 
     @Override
     public void execute(final ProjectInternal project) {
         if (LOGGER.isInfoEnabled()) {
             LOGGER.info("Evaluating {} using {}.", project, project.getBuildScriptSource().getDisplayName());
         }
+
+        File buildScriptSourceFile = project.getBuildscript().getSourceFile();
```

Comment:
According to doc, source file can be null if the script source is not a file. Not a file but what?

Reply:
It could be a URL. For example:

```Groovy
// build.gradle
apply from: 'https://gist.githubusercontent.com/bamboo/6cd380a0313a75eacda07be1b90fa722/raw/437a0d07e5c20919de9833001ff8846cb9278248/remote-script.gradle'
```

Reply:
@bamboo Here we could handle https://github.com/gradle/gradle/issues/22758

Reply:
After looking a bit more carefully at the problem I think it would be better to introduce a dedicated listener for script sources and move the event trigger to a single location in `ScriptPluginFactorySelector` so all possible ways of applying scripts are handled.

Here's how I'd start https://github.com/gradle/gradle/pull/22940